### DISCUSSION TITLE
Gracefully close scripts and server

### DISCUSSION THREAD

**Discussion by stive101** [Ym0T/pterodactyl-nginx-egg#18] [Q&A]
I have your egg running and see, that the server gets terminatet with ^C& ^C

Thats not shutting down the server, its just quit it, is ther a way to gracefully stop the server ? 

The issue for me is, im running a perl deamon, and tried everything to gracefully shutdown the deamon becouse otherwise it has problems in his function.

i put the start command in the nginx.sh

/path/to/deamon/deamon_run start

to stop the deamon gracfully, the server sould run 

/path/to/deamon/deamon_run stop

when u run it manualy, ^C also works to set the deamon into his shutdown state ( what means he cleans up temp files and remove his pid file, what e does not when he gets killd trough the Ptero panel stop command)

but i had no idea how i can execut that with the ptero stop container button ;(

thanks if u could help me :=)

**Comment by Ym0T** [comment]
Hey, I don't have a direct solution to your problem. However, I am currently trying and testing possible solutions.

**Comment by stive101** [comment]
Hi, 

Thanks for your fast relativ, 

^C wont shut down the deamon :/ it must be the comand /path/to/deamon/deamon_run stop
To shut it down gracefully, 
Did u know a way , how to execute a stop Script when u press stop in ptero , that my shut down my deamon ? And also nginx ? 

Nginx can be shut down with nginx -s stop that isnt working becouse he search into /run/nginx for the executable ,

Im  not a pro in doing such stuff

**Reply by Ym0T** [reply]
Sorry for the late reply. But I had tried a lot and couldn't find a reliable solution. Have you found anything yet?
